runner/runners/status_manager.go

```diff
@@ -130,6 +130,12 @@ func (s *StatusManager) Query(q runner.Query, wait runner.Wait) ([]runner.RunSta
 		return []runner.RunStatus{st}, s.svcStatus, nil
 	case <-timeout:
 		return nil, s.svcStatus, nil
+	case <-wait.AbortCh:
+		// we just consumed the abort request without really aborting the run,
+		// put a new request on the channel so taskRunner will abort the run.
+		wait.AbortCh <- 1
```

Comment:
We could make two copies of this chan in taskRunner (one for itself and one for query()) and then it could spin up a goroutine that listens to the original chan and then forwards the value to both new ones. Up to you whether it's worth doing this as we won't be adding another query implementation any time soon.